src/filter.rs

```diff
@@ -125,29 +163,35 @@ impl<D: Ord> KmerSummarizer<D, Vec<D>> for CountFilterSet<D> {
 /// * `stranded`: if true, preserve the strandedness of the input sequences, effectively
 ///   assuming they are all in the positive strand. If false, the kmers will be canonicalized
 ///   to the lexicographic minimum of the kmer and it's reverse complement.
-///
+/// * `report_all_kmers`: if true returns the vector of all the observed kmers and performs the
+///   kmer based filtering
+/// * `memory_size`: gives the size bound on the memory in GB to use and automatically determines
+///   the number of passes needed.
 /// # Returns
-/// Returns a tuple of valid kmers and invalid kmers. Valid kmers are a vector of
-/// (valid kmer, (extensions, summarized data)) tuples, invalid kmers are a vector of
-/// invalid kmers.
+/// BoomHashMap2 Object, check rust-boomphf for details
 #[inline(never)]
 pub fn filter_kmers<K: Kmer, V: Vmer<K>, D1: Clone, DS, S: KmerSummarizer<D1, DS>>(
     seqs: &[(V, Exts, D1)],
     summarizer: S,
     stranded: bool,
-) -> (Vec<(K, (Exts, DS))>, Vec<K>) {
+    report_all_kmers: bool,
+    memory_size: usize,
+)  -> ( boomphf::BoomHashMap2<K, Exts, DS>, Vec<K> )
+where DS: Debug{
```

Comment:
We shouldn't need the Debug constraint.  (BoomHashMap shouldn't need it)

Reply:
I think this bound is required by BoomHashMap2 for the `fmt` trait.